nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,343 @@
+  managementFile = managementFormat.generate "config.json" cfg.managementConfig;
+in
+{
+  meta.maintainers = with maintainers; [ thubrecht ];
+
+  options.services.netbird-server = {
+    enable = mkEnableOption (lib.mdDoc "netbird management service.");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.netbird;
+      defaultText = literalExpression "pkgs.netbird";
+      description = lib.mdDoc "The package to use for netbird";
+    };
+
+    dashboard = mkPackageOption pkgs "netbird-dashboard" { };
+
+    settings = mkOption {
+      type = with types; attrsOf (nullOr (oneOf [ (listOf str) bool int float str ]));
+      defaultText = ''
+        {
+          TURN_DOMAIN = cfg.settings.NETBIRD_DOMAIN;
+          TURN_PORT = 3478;
+          TURN_USER = "";
+          TURN_PASSWORD = null;
+
+          NETBIRD_SIGNAL_PROTOCOL = "http";
+          NETBIRD_SIGNAL_PORT = 443;
+
+          NETBIRD_AUTH_USER_ID_CLAIM = "sub";
+          NETBIRD_AUTH_CLIENT_SECRET = "";
+          NETBIRD_MGMT_API_CERT_FILE = null;
+          NETBIRD_MGMT_API_CERT_KEY_FILE = null;
+
+          NETBIRD_MGMT_IDP = "none";
+          NETBIRD_IDP_MGMT_CLIENT_ID = "";
+          NETBIRD_IDP_MGMT_CLIENT_SECRET = "";
+
+          NETBIRD_AUTH_SUPPORTED_SCOPES = [ "openid" "profile" "email" "offline_access" "api" ];
+          NETBIRD_AUTH0_DOMAIN = "";
+          NETBIRD_USE_AUTH0 = true;
+
+          NETBIRD_AUTH_DEVICE_AUTH_ENDPOINT = "";
+        }
+      '';
+      description = lib.mdDoc ''Configuration settings for netbird. List of strings [ a b ] will be concatenated as "a b"'';
+    };
+
+    managementConfig = mkOption {
+      inherit (managementFormat) type;
+      description = lib.mdDoc "Configuration of the netbird management server.";
+    };
+
+    idpManagerExtraConfig = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      description = lib.mdDoc "Extra options passed to the IdpManagerConfig.";
+    };
+
+    ports.management = mkOption {
+      type = types.port;
+      default = 8011;
+      description = lib.mdDoc "Internal port of the management server.";
+    };
+
+    ports.signal = mkOption {
+      type = types.port;
+      default = 8012;
+      description = lib.mdDoc "Internal port of the signal server.";
+    };
+
+    fetchOidcConfiguration = mkEnableOption "the automatic fetching of OIDC information.";
+
+    logLevel = mkOption {
+      type = types.enum [ "ERROR" "WARN" "INFO" "DEBUG" ];
+      default = "INFO";
+      description = lib.mdDoc "Log level of the netbird services.";
+    };
+
+    enableDeviceAuthorizationFlow = mkEnableOption "device authoriztion flow for netbird." // { default = true; };
+  };
+
+  config = mkIf cfg.enable {
+    services.netbird-server.settings = builtins.mapAttrs (_: mkDefault) {
+      TURN_DOMAIN = cfg.settings.NETBIRD_DOMAIN;
+      TURN_PORT = 3478;
+      TURN_USER = "";
+      TURN_PASSWORD = null;
+      TURN_SECRET = "secret";
```

Comment:
There should be a way to pass sensitive options via external file, all values passed to `services.netbird-server.settings` are going to be stored in nix store which is world-readable.

Reply:
I added `secretFiles` options for this